datafusion/common/src/dfschema.rs

```diff
@@ -656,9 +656,26 @@ impl DFSchema {
             (othertype, DataType::Dictionary(_, v1)) => v1.as_ref() == othertype,
             (DataType::List(f1), DataType::List(f2))
             | (DataType::LargeList(f1), DataType::LargeList(f2))
-            | (DataType::FixedSizeList(f1, _), DataType::FixedSizeList(f2, _))
-            | (DataType::Map(f1, _), DataType::Map(f2, _)) => {
-                Self::field_is_logically_equal(f1, f2)
+            | (DataType::FixedSizeList(f1, _), DataType::FixedSizeList(f2, _)) => {
+                // Don't compare the names of the technical inner field
+                // Usually "item" but that's not mandated
```

Comment:
that  depends what "logically equal" actually means.
The definition for this function is currently by-an-example, we should try to formalize what we mean.